python/composio/tools/toolset.py

```diff
@@ -1398,19 +1398,18 @@ def initiate_connection(
         :param auth_scheme: (Optional[AuthSchemeType]): Authentication scheme to use
         :return: (ConnectionRequestModel) Details of the connection request.
         """
-
-        if auth_scheme not in AUTH_SCHEMES:
+        if auth_scheme is not None and auth_scheme not in AUTH_SCHEMES:
             raise ComposioSDKError(f"'auth_scheme' must be one of {AUTH_SCHEMES}")
 
-        if auth_scheme is not None:
-            if auth_scheme not in AUTH_SCHEMES:
-                raise ComposioSDKError(f"'auth_scheme' must be one of {AUTH_SCHEMES}")
-
         if integration_id is None:
             if app is None:
                 raise ComposioSDKError(
                     message="Both `integration_id` and `app` cannot be None"
                 )
+
+            if auth_scheme is None:
+                auth_scheme = self.get_auth_scheme_for_app(app).auth_mode
```

Comment:
Consider adding error handling around `get_auth_scheme_for_app` call and a debug log when falling back to default auth scheme. This would help with troubleshooting:

```python
try:
    auth_scheme = self.get_auth_scheme_for_app(app).auth_mode
    self.logger.debug(f"Using default auth scheme {auth_scheme!r} for app {app.name!r}")
except Exception as e:
    raise ComposioSDKError(f"Failed to fetch default auth scheme for app {app.name!r}: {str(e)}")
```